parser.rb

```diff
@@ -0,0 +1,29 @@
+class ParserGamesLogs
+  attr_accessor :header_line 
+
+  def initialize(filename_with_path)
+    @filename_with_path = filename_with_path
+    @mode = 'r'
+    @header_line = read_first_line()
+    
+    unless is_header_valid?()
+      raise "Invalid file - The header does not match a valid pattern"
+    end
+  end
+  
+  private
+
+  def read_first_line()
+    File.open(@filename_with_path, @mode) do |log_file|
+      log_file.each_line do |line|
+        return line
+      end
+    end
```

Comment:
A iteração pelo arquivo é desnecessária nesse caso de querer retornar apenas a primeira linha. Kudos por usar o `each_line` ao invés do `readlines.each`

```suggestion
    File.open(@filename_with_path, @mode) do |log_file|
      log_file.readline
    end
```